### PR TITLE
libbpf-tools/memleak: fix 'show allocs'

### PR DESCRIPTION
**Pull request by EspenG**

### CODE REVIEW

**Comment by yonghong-song** [comment]
@inwardvessel could you help take a look at this patch?

File: libbpf-tools/memleak.c [hunk 1]
```diff
@@ -644,6 +651,14 @@ int print_stack_frames(struct allocation *allocs, size_t nr_allocs, int stack_tr
 
 		printf("%zu bytes in %zu allocations from stack\n", alloc->size, alloc->count);
 
+		if (env.show_allocs) {
+			struct allocation_node* it = alloc->allocations;
+			while (it != NULL) {
+				printf("\taddr = %#lx size = %ld\n", it->address, it->size);
```
Comment (inwardvessel) [inline libbpf-tools/memleak.c:657]: minor: maybe use `%zu` for `it->size`?

File: libbpf-tools/memleak.c [hunk 2]
```diff
@@ -773,6 +788,14 @@ int print_outstanding_allocs(int allocs_fd, int stack_traces_fd)
 				alloc->size += alloc_info.size;
 				alloc->count++;
 
+				if (env.show_allocs) {
+					struct allocation_node* node = malloc(sizeof(struct allocation_node));
```
Comment (inwardvessel) [inline libbpf-tools/memleak.c:792]: could add a check to make sure `malloc()` worked

File: libbpf-tools/memleak.c [hunk 3]
```diff
@@ -783,12 +806,21 @@ int print_outstanding_allocs(int allocs_fd, int stack_traces_fd)
...
+		struct allocation alloc = {
 			.stack_id = alloc_info.stack_id,
 			.size = alloc_info.size,
-			.count = 1
+			.count = 1,
+			.allocations = NULL
 		};
 
+		if (env.show_allocs) {
+			struct allocation_node* node = malloc(sizeof(struct allocation_node));
```
Comment (inwardvessel) [inline libbpf-tools/memleak.c:817]: same with a check here

**Review by inwardvessel — APPROVED** [review]
looks good. I tested locally and all seems good. thanks for adding this. some minor comments.

**Comment by EspenG** [comment]
@inwardvessel added fixes to your comments

**Comment by inwardvessel** [comment]
@yonghong-song good to merge

**Pull request merged**